pkg/generate/app/imageref.go

```diff
@@ -104,7 +104,10 @@ func (g *imageRefGenerator) FromStream(stream *imageapi.ImageStream, tag string)
 		return nil, err
 	}
 
-	if len(tag) == 0 && len(ref.Tag) == 0 {
+	switch {
+	case len(tag) > 0:
+		ref.Tag = tag
```

Comment:
because the tag that matched whatever it was that you were looking for is in the ComponentMatch.imageTag which is what gets passed here https://github.com/openshift/origin/blob/master/pkg/generate/app/imagelookup.go#L344